### PR TITLE
Call eglPresentationTimeANDROID when available.

### PR DESCRIPTION
**Pull request by tpsiaki**

### CODE REVIEW

**Comment by tpsiaki** [comment]
It's been my experience that it works best for the app to be able to set the frame time so that at the other end of MediaCodec, frames can still be identified with what was rendered in them.  For example, in an AR app, one might want the camera's capture time used as the timestamp, which wouldn't necessarily correlate with the wall clock time in the renderer.  Of course, doing it that way would be significantly more plumbing.

File: filament/src/driver/opengl/OpenGLDriver.cpp [hunk 1]
```diff
@@ -2488,6 +2488,7 @@ void OpenGLDriver::beginFrame(uint64_t monotonic_clock_ns, uint32_t frameId) {
             }
         }
     }
+    mContextManager.setPresentationTime(monotonic_clock_ns);
```
Comment (pixelflinger) [inline filament/src/driver/opengl/OpenGLDriver.cpp]: I don't know if this clock is correct. What's passed here is the steady_clock and I don't think this one has anything to do with the system time. What clock does this API expect?

Wouldn't it be easier to NOT add a new method and instead just do it where we do eglSwapBuffers() and use the system time?

**Comment by romainguy** [comment]
`ContextManagerCocoa` and `ContextManagerWebGL` are missing `setPresentationTime()`.

**Comment by pixelflinger** [comment]
I don't think we need the extra flexibility at this point. I'm not
confident this API is what we want. So I prefer to do something automatic
that doesn't add new apis for now.

**Comment by tpsiaki** [comment]
Added WebGL & Cocoa stubs.

I also added EGL_RECORDABLE_ANDROID to the config which is the last part needed to make this work with MediaCodec surfaces.

File: filament/src/driver/opengl/ContextManagerEGL.cpp [hunk 2]
```diff
@@ -537,6 +537,7 @@ std::unique_ptr<Driver> ContextManagerEGL::createDriver(void* const sharedGLCont
             EGL_GREEN_SIZE,  8,
             EGL_BLUE_SIZE,   8,
             EGL_ALPHA_SIZE,  0, // reserved to set ALPHA_SIZE below
+            EGL_RECORDABLE_ANDROID, 1,
```
Comment (pixelflinger) [inline filament/src/driver/opengl/ContextManagerEGL.cpp]: That's okay, but if we can't find a config with RECORDABLE_ANDROID, then we need to try again without it (and maybe print a warning in the log).

Comment (tpsiaki) [reply filament/src/driver/opengl/ContextManagerEGL.cpp]: done.

File: filament/src/driver/opengl/ContextManagerEGL.cpp [hunk 3]
```diff
@@ -63,6 +63,9 @@ using namespace utils;
 namespace filament {
 using namespace driver;
 
+using PFNEGLGETNATIVECLIENTBUFFERANDROIDPROC =
```
Comment (pixelflinger) [inline filament/src/driver/opengl/ContextManagerEGL.cpp]: Why was this added?

**Comment by tpsiaki** [comment]
PFNEGLGETNATIVECLIENTBUFFERANDROIDPROC was giving me an error when building for android locally, but presumably it's a setup thing since it works on kokoro without.  Removed.
